aten/src/ATen/native/native_functions.yaml

```diff
@@ -105,13 +112,15 @@
   named_guard: False
 
 - func: abs_(Tensor(a!) self) -> Tensor(a!)
+  exclude_from_c10_dispatcher: True
```

Comment:
I think you forgot to mention in the description that inplace and _out ops can't be registered at the moment because of Tensor & in their signature. Did you decide on an approach for this yet btw?

Reply:
no. out functions don't work anyhow because jit and c++ have a different argument order. Not sure yet about how to fix this.

Reply:
according to @gchanan, the reason for is that _out variants of functions have `out` argument as `kwarg_only` in python, but in C++ it's logical to always provide it while still have other args with default values. Thus the only option is to make it the first argument

Relevant code seems to be:
https://github.com/pytorch/pytorch/blob/33a1c30cb117c2e39126e49b5ec4a2b9b7a64b0d/aten/src/ATen/native_parse.py#L190
https://github.com/pytorch/pytorch/blob/33a1c30cb117c2e39126e49b5ec4a2b9b7a64b0d/tools/jit/gen_jit_dispatch.py#L452

Note that all C++ APIs today: public API, kernel implementations all use "reordered" variant with `out` argument first. TorchScript parsing and the Stack in JIT uses the "original" order.

If we try to introduce reordering logic in c10 it'd cause a mess when switching between boxed and unboxed APIs.

I see several options:
1) switch kernel definitions to "original" order (giant codemod) and keep reordering only in generated C++ API exposed to user
2) try mirroring reordering logic in c10 such that all boxing/unboxing functions do reordering
3) if we give up the idea of "no codegen" (which I think is reasonable in mid-term). Then for native_functions we can provide ability to codegen boxing/unboxing wrappers and just register them with c10 as hooks (and utilize current codegen). Default hook would be the regular templated wrappers generated automatically based on C++ signature for cases where there's no reordering.

Reply:
I'd view this problem as analogous to the TensorOptions issue.  There, we do some codegen in order to make the C++ API nice (good!), but then need to reimplement the crazy logic in multiple places (bad!).  All of these endpoints don't need to use the fancy C++ signature, they aren't user endpoints, a dumb signature is totally fine.

Now, you may or may not want a _single_ way of telling downstream components that they should actually map to a different C++ function call so we could have private implementation calls (which would allow us to e.g. only expose a single "public" function for factories and out functions, as we do now), but that logic can be super simple.